### ISSUE TITLE
Subfolder string is used as scheme in generating sitemap error message.

### ISSUE THREAD

**Issue by Hailong** [magento/magento2#24623]
### Preconditions (*)

1. Magento 2.3.2 & 2.3-develop
2. Make sure PR https://github.com/magento/magento2/pull/24530 is applied.
3. Web folder is `/pub`
4. Mac OS
### Steps to reproduce (*)

1. Install Magento 2.3.2 or 2.3-develop on Linux or Unix machine.
2. Create a subfolder named `site` under Magento root directory and change its ownership to other user account, so that the web application account does not have writable permission.
3. Go to Magento backend, and navigate to Marketing -> Site Map
4. Add a sitemap with Path = '/site' and Filename = 'sitemap.xml'
5. Click "Generate" to generate the sitemap.

### Expected result (*)

Error message saying the directory is not writable.

### Actual result (*)

Would get the error message looks like following:
```    
The path "/site:///Users/hailong/dev/magento2_my_dev/" is not writable.
```
<img width="1158" alt="image" src="https://user-images.githubusercontent.com/354776/65057104-10881980-d940-11e9-9870-d2d9ee257ee0.png">
The problem here is the `site` string was taken as scheme string. There must be some problem in the file system related code.

**Comment by engcom-Charlie** [comment]
Hello @Hailong 
This issue will be finally confirmed if your PR is fully reviewed and confirmed.

I have the same result on the vanilla Magento 2.3-develop and your PR:

![image](https://user-images.githubusercontent.com/51680850/65051158-4b567580-d971-11e9-9034-a152aeb7eb32.png)
As I understand it is the expected result or not?
Could you take a look?

**Comment by Hailong** [comment]
Hi @engcom-Charlie , here is what I got from my side. I just merged in the latest 2.3-develop branch, and with PR https://github.com/magento/magento2/pull/24530 together.

My environment is Mac OS with Nginx, and the web folder is `/pub`.

<img width="1158" alt="image" src="https://user-images.githubusercontent.com/354776/65057104-10881980-d940-11e9-9870-d2d9ee257ee0.png">

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-20220` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @Hailong. Thank you for your report. 
The issue has been fixed in magento/magento2#24675 by @Hailong in 2.3-develop branch
Related commit(s):
 - [3adb1eb96d6427e4d4c91aad65eac38aea77b8f5](https://github.com/magento/magento2/commit/3adb1eb96d6427e4d4c91aad65eac38aea77b8f5)

The fix will be available with the upcoming 2.3.4 release.
